data/data-pipeline/data_pipeline/comparison_tool/src/utils.py

```diff
@@ -0,0 +1,330 @@
+import pandas as pd
+from data_pipeline.score import field_names
+
+
+FIPS_MAP = {
+    "53": "WA",
+    "10": "DE",
+    "11": "DC",
+    "55": "WI",
+    "54": "WV",
+    "15": "HI",
+    "12": "FL",
+    "56": "WY",
+    "72": "PR",
+    "34": "NJ",
+    "35": "NM",
+    "48": "TX",
+    "22": "LA",
+    "37": "NC",
+    "38": "ND",
+    "31": "NE",
+    "47": "TN",
+    "36": "NY",
+    "42": "PA",
+    "02": "AK",
+    "32": "NV",
+    "33": "NH",
+    "51": "VA",
+    "08": "CO",
+    "06": "CA",
+    "01": "AL",
+    "05": "AR",
+    "50": "VT",
+    "17": "IL",
+    "13": "GA",
+    "18": "IN",
+    "19": "IA",
+    "25": "MA",
+    "04": "AZ",
+    "16": "ID",
+    "09": "CT",
+    "23": "ME",
+    "24": "MD",
+    "40": "OK",
+    "39": "OH",
+    "49": "UT",
+    "29": "MO",
+    "27": "MN",
+    "26": "MI",
+    "44": "RI",
+    "20": "KS",
+    "30": "MT",
+    "28": "MS",
+    "45": "SC",
+    "21": "KY",
+    "41": "OR",
+    "46": "SD",
+}
+
+
+def read_file(
+    file_path: str, columns: list, geoid: str = field_names.GEOID_TRACT_FIELD
+) -> pd.DataFrame:
+    """Reads standardized csvs in
+
+    Parameters:
+        file_path: the file to read
+        columns: the columns to include
+        geoid: the geoid column name (if we change this in usa.csv, will need to change this slightly)
+
+    Returns:
+        dataframe that has been read in with geographic index
+    """
+    assert (
+        geoid == field_names.GEOID_TRACT_FIELD
+    ), f"Field name specified for geoid is incorrect. Use {field_names.GEOID_TRACT_FIELD}"
+    return pd.read_csv(
+        file_path, usecols=columns + [geoid], dtype={geoid: str}
+    ).set_index(geoid)
+
+
+def produce_summary_stats(
+    joined_frame: pd.DataFrame,
+    comparator_column: str,
+    score_column: str,
+    population_column: str,
+    geoid_column: str = field_names.GEOID_TRACT_FIELD,
+) -> pd.DataFrame:
+    """Produces high-level overview dataframe
+
+    Parameters:
+        joined_frame: the big df
+        comparator_column: the column name for the comparator identification bool
+        score_column: the column name for the CEJST score bool
+        population_column: the column that includes population count per tract
+        geoid_column: the geoid10_tract column
+
+    Returns:
+        population: the high-level overview df
+    """
+    population = joined_frame.groupby([comparator_column, score_column]).agg(
+        {population_column: ["sum"], geoid_column: ["count"]}
+    )
+
+    population["share_of_tracts"] = (
+        population[geoid_column] / population[geoid_column].sum()
+    )
+
+    population["share_of_population_in_tracts"] = (
+        population[population_column] / population[population_column].sum()
+    )
+
+    population.columns = [
```

Comment:
nitty little nit: directly renaming columns is a little less safe than using the `rename` method and specifying a dictionary from old to new names. 

I know I can be annoying about some of this stuff, but I'll elaborate a bit on these examples of how I think about it, in case that's helpful. I'm always thinking about the codebase as a product on its own -- how easy is it to use and understand? How easy is it to make mistakes in the codebase? With this direct renaming, it will break if anyone changes the order of the lines above, adds a new field, deletes a field, etc, and the way in which it breaks will not make it obvious what changed to break it. 

It's also hard for me to verify that these renamings in this version are in the right order unless I open up the code and run it myself and see what the 4 columns contain. 

My dear old friend @dfkoh used to always emphasize, the goal of resilient software engineering is to increase the number of mistakes someone has to make for something to go really wrong. Many software codebases/organizations are just one mistake organizations: someone only needs to hit one wrong button for bad things to happen. In the safest codebases on earth, you're probably looking at 3 mistakes for most problems, 2 for some. 

Here, using the renaming function instead will make it so one mistake (e.g., changing order of the code that creates fields above) doesn't cause a break. Even better, if you use `.rename(..., errors='raise')`, that's the best, because then if a developer makes a mistake (e.g., still trying to rename a field that no longer exists), they'll immediately be alerted to the change and the way that error is raised will make it obvious what's happening (i.e., that the field no longer exists). 

Anyways that was a long rant but just felt compelled to share some thoughts on why I'm always so nitty 😅 .

Reply:
not too nitty! happy to implement.

Reply:
(just a head's up -- the code will be a little less lean because there's a multi-index from `agg`)